app/parse.py

```diff
@@ -15,8 +39,106 @@ class Product:
     num_of_reviews: int
 
 
+def get_product(product_card: WebElement) -> Product:
+    return Product(
+        title=product_card.find_element(By.CLASS_NAME, "title").get_attribute("title"),
+        description=product_card.find_element(By.CLASS_NAME, "description").text,
+        price=float(
+            product_card.find_element(By.CLASS_NAME, "price").text.replace("$", "")
+        ),
+        rating=len(product_card.find_elements(By.CLASS_NAME, "ws-icon-star")),
+        num_of_reviews=int(
+            product_card.find_element(By.CLASS_NAME, "review-count").text.split()[0]
+        ),
+    )
+
+
+def get_all_from_dynamic_page(url: str) -> list[Product]:
+    logging.info(f"Get all products dynamic page'{url}'")
+    driver = get_driver()
+    driver.get(urljoin(BASE_URL, url))
+    try:
+        close_button = driver.find_element(By.CLASS_NAME, "acceptCookies")
+        driver.execute_script("arguments[0].scrollIntoView(true);", close_button)
+        close_button.click()
+        logging.info("Closed cookie banner")
+    except NoSuchElementException:
+        logging.info("Cookie banner not found")
+    except ElementNotInteractableException:
+        logging.info("The cookie banner was unclickable")
+    try:
+        more = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+        while True:
+            try:
+                if not more.is_displayed():
+                    break
```

Comment:
Consider adding a maximum number of attempts or a timeout to prevent an infinite loop in case the 'Load More' button remains visible but is not functional.